### ISSUE TITLE
Add more comments in the source code

### ISSUE THREAD

**Issue by Satinelamp** [Ionizing/rsgrad#3]
Maybe it would be more clear and more understandable if there are more comments in the source code, especially files in the vasp_parser directory.

**Comment by Ionizing** [comment]
You're right. I did it via TDD (test driven development), which means the tests are the best comments. With the benefit of Rust file organization, the tests are bundled with the source code. For example, the tests for Outcar class are on the bottom of outcar.rs and they are very helpful to understand the code. I tried my best to decouple every method and if you have any questions on them, just reply here 😁 and I'll consider adding comments for them.
